### PR TITLE
feat(pins): add support for custom storage options in `board_s3`

### PR DESCRIPTION
**Pull request by ericmjl**
- Introduce the ability to pass arbitrary storage options to the underlying fsspec S3FileSystem in `board_s3`.
- This enhancement allows for greater flexibility when connecting to S3-compatible services by enabling the specification of custom credentials, endpoints, and other S3FileSystem parameters.
- Added documentation and examples to illustrate how to use the new `storage_options` parameter, including an example for connecting to Backblaze B2.

This change enables users to more easily integrate with a variety of S3-compatible storage solutions, improving the library's versatility and user experience.

Address issue: #235

Tag @isabelizimm!

### CODE REVIEW

**Comment by ericmjl** [comment]
@isabelizimm @machow I think I'm running into an issue with the docs, but I'm not quite sure how to address the issue, as it uses quartodoc (which I'm happy to learn, but may need help getting up-to-speed on).

In case it's helpful, this is a [direct link](https://github.com/rstudio/pins-python/actions/runs/9896203250/job/27340731320?pr=237#step:6:159) to the exact line that is failing.

**Comment by isabelizimm** [comment]
Ah thank you so much for putting up this PR! Let me peek at the docs failures and run the tests on other backends to see if something jumps out  👀

File: pins/constructors.py [hunk 1]
```diff
@@ -511,19 +513,36 @@ def board_s3(path, versioned=True, cache=DEFAULT, allow_pickle_read=None):
...
+    ...     secret=os.getenv("FSSPEC_S3_SECRET"),
+    ... )
 
     See <https://github.com/fsspec/s3fs>
 
     """
     # TODO: user should be able to specify storage options here?
 
     opts = {"listings_expiry_time": 0}
+    opts.update(**storage_options)
```
Comment (isabelizimm) [inline pins/constructors.py:545]: We should probably check `storage_options` and raise a warning if people overwrite the `listings_expiry_time`. It's okay to not error/fully block people from overwriting it, but just note there may be surprising behavior on `pin_read` if this value is not 0. 


Info on the [listings cache](https://filesystem-spec.readthedocs.io/en/latest/features.html#listings-caching) that explains what `listings_expiry_time` does, if you're interested.

**Comment by isabelizimm** [comment]
One small nit, but everything else looks great! 😄

**Comment by ericmjl** [comment]
Woot! Looks like all of the non-skipped stuff passed :smile:.

File: pins/constructors.py (hunk 1 above)
Comment (ericmjl) [reply pins/constructors.py:545]: On it, I'll get to it right away.

**Comment by ericmjl** [comment]
Okie dokes, just pushed the requested change, @isabelizimm!

**Review by isabelizimm — APPROVED** [review]
Spectacular! This is so great to expand the options for S3 pins users 🔥

**Pull request merged**